### PR TITLE
feat: add file searching to Code browser

### PR DESCRIPTION
**Pull request by liangfung**
Close TAB-555

https://jam.dev/c/60fb09c8-ea11-4cd0-85bc-ab133ab09c1c

### CODE REVIEW

**Comment by wsxiaoys** [comment]
cc @boxbeam to use as a reference for api design

**Comment by wsxiaoys** [comment]
shall we also adding the `t` shortcut to focus on the file search box?

image

**Comment by boxbeam** [comment]
What are the proposed endpoints? I can't figure it out from the code

**Comment by wsxiaoys** [comment]
> What are the proposed endpoints? I can't figure it out from the code

it's not in the PR. The API design is in linear issue.

**Comment by boxbeam** [comment]
I don't see any API design specified in the Linear issue, just a Jam link which shows its usage. Checked in the network tab and don't see anything. What endpoints do we need?

File: ee/tabby-ui/app/files/components/file-tree-header.tsx [hunk 1]
```diff
@@ -13,61 +28,366 @@ import {
...
+                <ComboboxAnchor>
+                  <div className="relative">
+                    <ComboboxInput
+                      className="pr-8"
+                      // placeholder="Go to file"
+                      placeholder={
+                        repoName
+                          ? 'Go to file'
+                          : 'Go to file (pick a repository first)'
+                      }
```
Comment (wsxiaoys) [inline ee/tabby-ui/app/files/components/file-tree-header.tsx:245]:
```suggestion
                      placeholder="Go to file"
```

**Pull request merged**